server.go

```diff
@@ -2475,9 +2475,17 @@ func setupRPCListeners() ([]net.Listener, error) {
 	// Setup TLS if not disabled.
 	listenFunc := net.Listen
 	if !cfg.DisableRPC && !cfg.DisableTLS {
-		// Generate the TLS cert and key file if both don't already
-		// exist.
-		if !fileExists(cfg.RPCKey) && !fileExists(cfg.RPCCert) {
+		// Generate the TLS cert and key file if both don't already exist.
+		keyFileExists := fileExists(cfg.RPCKey)
+		certFileExists := fileExists(cfg.RPCCert)
+		if len(cfg.AltDNSNames) != 0 && (keyFileExists || certFileExists) {
+			rpcsLog.Warn("Additional DNS names specified when TLS " +
+				"certificates already exist will NOT be included:")
+			rpcsLog.Warnf("- In order to create TLS certs that include the "+
+				"additional DNS names, delete %q and %q and restart the server",
+				cfg.RPCKey, cfg.RPCCert)
+		}
+		if !keyFileExists && !certFileExists {
```

Comment:
This isn't a change of logic, but I've previously wondered why this isn't an `||`.  If either key or cert do not exist, how can the RPC server be expected to start?

Reply:
If one is missing the call just below to `tls.LoadX509KeyPair` will fail with an obvious missing file message.  This check is an `&&` so that it only attempt to generate new certificates when neither file exists since if one or the other exists, the user probably has done something wonky and we don't want to just wipe out their existing file blindly.